plugin/core/windows.py

```diff
@@ -739,3 +756,145 @@ def _format_request(self, direction: str, method: str, request_id: Any) -> str:
 
     def _format_notification(self, direction: str, method: str) -> str:
         return "{} {} {}".format(direction, self._server_name, method)
+
+
+class RemoteLogger(Logger):
+    PORT = 9981
+    DIRECTION_OUTGOING = 1
+    DIRECTION_INCOMING = 2
+    _ws_server = None  # type: Optional[WebsocketServer]
+    _ws_server_thread = None  # type: Optional[threading.Thread]
+
+    def __init__(self, manager: WindowManager, server_name: str) -> None:
+        self._manager = ref(manager)
+        self._server_name = server_name
+        if not RemoteLogger._ws_server:
+            try:
+                RemoteLogger._ws_server = WebsocketServer(self.PORT)
+                RemoteLogger._ws_server.set_fn_new_client(self._on_new_client)
+                RemoteLogger._ws_server.set_fn_client_left(self._on_client_left)
+                RemoteLogger._ws_server.set_fn_message_received(self._on_message_received)
+            except OSError as ex:
+                if ex.errno == 48:  # Address already in use
+                    debug('WebsocketServer not started - address already in use')
+                    RemoteLogger._ws_server = None
+                else:
+                    raise ex
+        self._start_server()
+
+    def _start_server(self) -> None:
+        if RemoteLogger._ws_server:
+            def start_async() -> None:
+                if RemoteLogger._ws_server:
+                    RemoteLogger._ws_server.run_forever()
+            RemoteLogger._ws_server_thread = threading.Thread(target=start_async)
+            RemoteLogger._ws_server_thread.start()
+
+    def _stop_server(self) -> None:
```

Comment:
Should this not be called in a `__del__` somewhere?

Reply:
Since it's initalialized once per however many sessions, it can't be stopped on any particular session being destroyed. It could only be stopped when all active sessions have been stopped but I'm not sure how to hook into that. Probably would require your session changes anyway?